cmd/kube-proxy/app/server.go

```diff
@@ -287,6 +312,40 @@ func (o *Options) Validate(args []string) error {
 	return nil
 }
 
+// ValidateMixedArguments validates passed arguments.
+func (o *Options) ValidateMixedArguments(flag *pflag.FlagSet) error {
+	// If --config isn't set, we have nothing to validate
+	if !flag.Changed("config") {
+		return nil
+	}
+
+	mixedInvalidFlags := []string{}
+	flag.Visit(func(f *pflag.Flag) {
+		if isAllowedFlag(f.Name) {
+			// allowed flags can be set with --config
+			return
+		}
+		mixedInvalidFlags = append(mixedInvalidFlags, f.Name)
+	})
+
+	if len(mixedInvalidFlags) != 0 {
+		return fmt.Errorf("can not mix '--config' with arguments %v", mixedInvalidFlags)
+	}
+	return nil
+}
+
+func isAllowedFlag(flagName string) bool {
+	knownFlags := sets.NewString(KubeProxyConfig,
```

Comment:
It's ineffient to regenerate this for every call to `isAllowedFlag`. I think this simplest would be to just drop this function and create the `knownFlags` set in `ValidateMixedArguments` before the loop. (Though it needs a better name. `allowedMixedFlags`?)

Also, I don't think having `const`s for these flags really improves things. Just inline the strings.